File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
import collections

def solution(participant, completion):
    # collections.Counter() -> 리스트 안의 각 요소가 몇 번 등장했는지 -> {이름: 개수}
    
    part_counter = collections.Counter(participant)
    comp_counter = collections.Counter(completion)
    
    answer = part_counter - comp_counter
        
    return list(answer.keys())[0]

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string, int> hash;
    
    for (auto& p : participant)
        hash[p]++;
    
    for (auto& c : completion)
        hash[c]--;
    
    for (auto& entry : hash)
        if (entry.second > 0)
            return entry.first;
}